streams/src/test/java/org/apache/kafka/streams/AutoOffsetResetTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams;
+
+import org.apache.kafka.streams.internals.AutoOffsetResetInternal;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class AutoOffsetResetTest {
+
+    @Test
+    void noneShouldReturnAnEmptyDuration() {
+        final AutoOffsetResetInternal none = new AutoOffsetResetInternal(AutoOffsetReset.none());
+        assertThrows(NoSuchElementException.class, none::duration, "None should not have a duration.");
+    }
+
+    @Test
+    void latestShouldReturnAnEmptyDuration() {
+        final AutoOffsetResetInternal latest = new AutoOffsetResetInternal(AutoOffsetReset.latest());
+        assertThrows(NoSuchElementException.class, latest::duration, "Latest should not have a duration.");
+    }
+
+    @Test
+    void earliestShouldReturnAnEmptyDuration() {
+        final AutoOffsetResetInternal earliest = new AutoOffsetResetInternal(AutoOffsetReset.earliest());
```

Comment:
nit: these 3 tests have the name `xShouldReturnAnEmptyDuration()` but the actual behavior is to throw a `NoSuchElementException` (which is expected by calling `get` on an empty `Optional`) - it's a little confusing it made my think it would return `Optional.empty()`